spec/features/posts_index_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe 'Post Index Page', type: :feature do
+  describe 'Viewing Post Index page' do
+    before(:each) do
+      @user1 = User.create(name: 'Tom', photo: 'https://unsplash.com/photos', bio: 'Teacher from Mexico.',
+                           post_counter: 0)
+      @first_post = Post.create(author: @user1, title: 'post1', text: 'This is my first post', likes_counter: 0,
+                                comments_counter: 0)
+      @comment1 = Comment.create(post: @first_post, author: @user1, text: 'Hi Tom!, Nice comment')
+      visit user_posts_path(@user1) # Use the named route 'user_posts_path' to visit the posts index page for the user
+    end
+
+    it 'should show user name' do
+      expect(page).to have_content(@user1.name)
+    end
+
+    it 'should show user photo' do
+      expect(page.html).to include(@user1.photo)
+    end
+
+    it 'should show post_counter of clicked user' do
+      expect(page).to have_content(@user1.post_counter)
+    end
+
+    it 'should show all posts' do
+      expect(page).to have_content(@first_post.title)
+    end
+
+    it 'should show post text' do
+      expect(page).to have_content(@first_post.text)
+    end
+
+    it 'should show comment text' do
+      expect(page).to have_content(@first_post.text)
+    end
+
+    it 'should show comments counter' do
+      expect(page).to have_content(@first_post.comments_counter)
+    end
+
+    it 'should show likes counter' do
+      expect(page).to have_content(@first_post.likes_counter)
+    end
+
+    it 'should navigate to post show page when clicked on post' do
+      click_on @first_post.title
+      expect(current_path).to eq(current_path)
+    end
```

Comment:
Nice work so far. However, you are missing some of the tests. For instance `Tests that asserts you can see a section for pagination if there are more posts than fit on the view.` is missing here. Kindly ensure that all tests cases have been covered. :+1: